<sdf version='1.7'>
  <world name='default'>
    <plugin name="set_entity_state" filename="libgazebo_ros_state.so"/>
    <light name='sun' type='directional'>
      <cast_shadows>1</cast_shadows>
      <pose>0 0 10 0 -0 0</pose>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.5 0.1 -0.9</direction>
      <spot>
        <inner_angle>0</inner_angle>
        <outer_angle>0</outer_angle>
        <falloff>0</falloff>
      </spot>
    </light>
    <gravity>0 0 0</gravity>
    <magnetic_field>6e-06 2.3e-05 -4.2e-05</magnetic_field>
    <atmosphere type='adiabatic'/>
    <physics type='ode'>
      <max_step_size>0.001</max_step_size>
      <real_time_factor>1</real_time_factor>
      <real_time_update_rate>1000</real_time_update_rate>
    </physics>
    <scene>
      <ambient>0.9 0.9 0.9 1</ambient>
      <background>0.7 0.7 0.7 1</background>
      <shadows>1</shadows>
    <sky>
    <clouds>
      <speed>12</speed>
    </clouds>
    </sky>
    <!-- <fog>
      <type>linear</type>
      <start>10</start>
      <end>500</end>
      <density>1</density>
    </fog> -->
    </scene>
    <!-- <include>
      <uri>model://hong_kong</uri>
    </include> -->
    <model name="hong_kong">
      <include>
        <uri>model://hong_kong</uri>
        <name>hk_city</name>
      </include>
      <pose>0 0 0 0 0 0</pose>  
      <!-- <pose>0 -60 0 0 0 -0.73</pose>   -->
    </model>
    <!-- <model name="sphere">
      <pose>0 0 1 0 0 0</pose>
      <link name="sphere_link">
        <visual name="sphere_visual">
          <geometry>
            <sphere>
              <radius>2</radius>
            </sphere>
          </geometry>
          <transparency>0.5</transparency>
          <plugin name="sphere_plugin" filename="libSphereModelPlugin.so"/>
        </visual>
        <collision name='sphere_collision'>
          <laser_retro>0</laser_retro>
          <max_contacts>10</max_contacts>
          <pose>0 0 0 0 -0 0</pose>
          <geometry>
            <box>
              <size>0.02 .01 .01</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model> -->
    
    <model name="drone_0">
      <static>false</static>
      <pose>0 0 1 0 0 0</pose>
      <include>
        <uri>model://x500</uri>
        <name>drone_0</name>
      </include>
    </model>
    <model name="laser_0">
      <static>false</static>
      <pose>0 0 1 0 0 0</pose>
      <!-- <include>
        <uri>model://x500</uri>
        <name>drone_0</name>
      </include> -->

      <link name="laser_link">
        <visual name="laser_visual">
          <geometry>
            <box>
              <size>5 2 2</size>
            </box>
          </geometry>
          <material>
            <ambient>0 0 0 1</ambient>
            <diffuse>0 0 0 1</diffuse>
            <specular>0 0 0 0</specular>
            <emissive>1 1 0 1</emissive>
          </material>
        </visual>
        <collision name='laser_collision'>
          <laser_retro>0</laser_retro>
          <max_contacts>10</max_contacts>
          <pose>0 0 0 0 -0 0</pose>
          <geometry>
            <box>
              <size>0.02 .01 .01</size>
            </box>
          </geometry>
        </collision>
        <sensor name="skycam" type="wideanglecamera">
          <pose>0 0 2.1 0 -1.5707 0</pose>
          <camera>
            <horizontal_fov>3.1415</horizontal_fov>
            <image>
              <width>500</width>
              <height>500</height>
            </image>
            <clip>
              <near>0.1</near>
              <far>100</far>
            </clip>
            <lens>
              <type>stereographic</type>
              <scale_to_hfov>true</scale_to_hfov>
              <cutoff_angle>3.1415</cutoff_angle>
              <env_texture_size>512</env_texture_size>
            </lens>
            <distortion>
              <k1>0.0</k1>
              <k2>0.0</k2>
              <k3>0.0</k3>
              <p1>0.0</p1>
              <p2>0.0</p2>
            </distortion>
          </camera>
          <always_on>1</always_on>
          <update_rate>1</update_rate>
          <plugin name="skycam_controller" filename="libgazebo_ros_camera.so">
            <!-- <ros>
              <namespace >skycam</namespace>
            </ros> -->
            <hack_baseline>0.07</hack_baseline>
          </plugin>
        </sensor>
        <!-- ray sensor -->
        <sensor name="sensor_ray" type="ray">
          <visualize>true</visualize>
          <ray>
            <scan>
              <horizontal>
                <samples>62</samples>
                <resolution>1</resolution>
                <min_angle>0</min_angle>
                <max_angle>0</max_angle>
              </horizontal>
              <vertical>
                <samples>1</samples>
                <resolution>1</resolution>
                <min_angle>0.8</min_angle>
                <max_angle>0.8</max_angle>
              </vertical>
            </scan>
            <range>
              <min>0.05</min>
              <max>50.0</max>
            </range>
          </ray>
          <always_on>true</always_on>
          <update_rate>100.0</update_rate>
          <plugin name="laserscan" filename="libGNSSMultipathSensorPlugin.so">
            <ros>
              <namespace>/gnss_multipath_plugin</namespace>
              <remapping>~/out:=scan</remapping>
            </ros>
            <output_type>sensor_msgs/LaserScan</output_type>
            <disableNoise>0</disableNoise>
            <origin_latitude>22.298251</origin_latitude>
            <origin_longitude>114.178761</origin_longitude>
            <origin_altitude>0</origin_altitude>
          </plugin>
        </sensor>
        <self_collide>0</self_collide>
      </link>
    </model>
    <!-- <gui>
      <camera name="user_camera">
        <projection_type>orthographic</projection_type>
        <pose>1.5 -1.25 0 0 1.578 0</pose>
      </camera>
    </gui> -->
    <gui fullscreen='0'>
      <camera name='user_camera'>
        <pose>0 0 100 0 1.57 0</pose>
        <view_controller>orbit</view_controller>
        <projection_type>orthographic</projection_type>
        <!-- <track_visual>
          <name>virtual_cam</name>
          <follow>false</follow>
          <xyz>-10 10 10 </xyz>
          <static>false</static>
          <min_dist>1.0</min_dist>
          <max_dist>5000.0</max_dist>
        </track_visual> -->
      </camera>
    </gui>
  </world>
</sdf>
<!-- vim: set et ft=xml fenc=utf-8 ff=unix sts=0 sw=2 ts=2 : -->
